spec/components/CioPlpGrid/CioPlpGrid.test.jsx

```diff
@@ -1,19 +1,43 @@
+/* eslint-disable max-classes-per-file */
 import React from 'react';
 import { render, waitFor } from '@testing-library/react';
+import '@testing-library/jest-dom';
 import CioPlpGrid from '../../../src/components/CioPlpGrid';
 import CioPlp from '../../../src/components/CioPlp';
 import { DEMO_API_KEY } from '../../../src/constants';
-import '@testing-library/jest-dom';
-import { transformSearchResponse } from '../../../src/utils/transformers';
-import mockSearchResponse from '../../local_examples/apiSearchResponse.json';
+import mockApiSearchResponse from '../../local_examples/apiSearchResponse.json';
+import mockApiBrowseResponse from '../../local_examples/apiBrowseResponse.json';
+import { transformBrowseResponse, transformSearchResponse } from '../../../src/utils/transformers';
 import { RequestStatus } from '../../../src/components/CioPlpGrid/reducer';
+import useCioPlp from '../../../src/hooks/useCioPlp';
+import useRequestConfigs from '../../../src/hooks/useRequestConfigs';
+
+const actualUseCioPlp = jest.requireActual('../../../src/hooks/useCioPlp').default;
+const actualUseRequestConfigs = jest.requireActual('../../../src/hooks/useRequestConfigs').default;
+
+jest.mock('../../../src/hooks/useCioPlp');
+jest.mock('../../../src/hooks/useRequestConfigs');
+jest.mock('@constructor-io/constructorio-client-javascript/lib/modules/search.js', () => {
+  const Search = class {
+    // eslint-disable-next-line @typescript-eslint/no-useless-constructor, @typescript-eslint/no-empty-function
+    constructor() {}
+
+    getSearchResults = jest.fn().mockResolvedValue(mockApiSearchResponse);
+  };
```

Comment:
We could do something like the code below instead, but I found this version more readable even though it's slightly more verbose
```
jest.mock('@constructor-io/constructorio-client-javascript/lib/modules/search.js', () =>
  jest.fn().mockImplementation(() => ({
    getSearchResults: jest.fn().mockResolvedValue(mockApiSearchResponse),
  })),
);
```

Reply:
I also opted to mock the modules themselves rather than the whole client to keep the mocking to a minimum.